lib/child_process.js

```diff
@@ -467,7 +467,10 @@ function normalizeSpawnArguments(file, args, options) {
   var envPairs = [];
 
   for (var key in env) {
```

Comment:
I think this should optimally be `const envPairs = Object.keys(env).filter(Boolean).map(key => `${key}=${env[key]}`)` or something similar (ignore empty keys in addition to undefined ones.

I do think it's underspecified though. So I'll wait for someone who understands it better to weigh in.

Reply:
OMG `.filter(Boolean)` 💘 
I used to do `.filter((x) => x)`, but have alway looked for a built-in.

P.S. @Gerhut in TF&I (node 8.3.0 and newer) `for ... of` loop are almost 10 time faster than `for ... in` loops, and it's supports `Object.entries()`

Reply:
@refack but the call to `Object.keys` is itself about as slow as doing `for... in` - in this particular case I don't think performance is important given the time it takes to create a process and all 😅

Reply:
@benjamingr agreed.
I'm just promoting TF&I good behaviour (no more CranksharfScript)

Reply:
Since the code around still in legacy Syntax, should I be allowed to use modern syntax? If so I will modify them in modern syntax.

Reply:
@Gerhut yes, in general new code should be written in new syntax.

Reply:
However, `for..in` includes keys in prototype, and keys-in-prototype of env test exists at https://github.com/nodejs/node/blob/master/test/parallel/test-child-process-env.js#L31-L33 . Changing to `Object.keys/entries` will fail them

Reply:
I think it would be good to remove that test case and only use `Object.keys`. There is not much reasoning in the original commit about why it was added but I think this is actually a faulty behavior.

Reply:
Using `filter(Boolean)` would be a bad decision to use though as it would remove a empty string and any other falsy value as well.